netbox_netprod_importer/importer.py

```diff
@@ -246,7 +247,10 @@ def get_lldp_neighbours(self):
         assert self.device.device
 
         try:
-            yield from self.specific_parser.get_detailed_lldp_neighbours()
+            if self.discovery_protocol == "cdp":
```

Comment:
It is a little weird here to put both lldp and cdp in the same try/except with a catch of `NotImplementedError`.
I wrote that to fallback on bare napalm in case the specific parser would not implement a more advanced function to do the discover over lldp.

Here it seems weird if your specific parser doesn't implement a discover via cdp, as it will fallback on lldp on napalm without any error when the user specified to use CDP.

I would move the cdp and lldp part in their own functions, and raise a `ValueError("CDP discovery protocol wanted but not implemented by the underlying parser")` (or whatever) for the CDP inside the catch of a `NotImplementedError`.

Feel free to ask me some details if it's not clear, or if you prefer I will apply the change by myself when squashing for commits ;)

Reply:
In this case, I did not know how to properly implement (little experience) and saw that it was a crutch ... Therefore, you can explain and direct in the right direction, or if it is easier for you to redo this part, I will look and learn;)

Reply:
look at this [commit](https://github.com/Anthony25/netbox-netprod-importer/pull/15/commits/ec0d22dc7132ac418c3bb6179c2126aa214a2f2f)

Reply:
Yes, it looks great!